bufferoverride/gui/bufferoverrideview.cpp

```diff
@@ -58,6 +62,49 @@ bool BufferOverrideView::attached(VSTGUI::CView * parent) {
 }
 
 
+static std::tuple<CCoord, CCoord, CCoord>
+ScaleViewData(BufferOverrideViewData data, CCoord pixels_per_window) {
+
+  constexpr CCoord window_sec_default = 1;
+  auto window_sec = window_sec_default;
+
+  constexpr auto Scale = [](CCoord value, CCoord window) {
+    return value * (window_sec_default / window);
+  };
+
+  // calculate scale using the pre-LFO values so that it remains
+  // stable when not adjusting parameters
+  const CCoord forced_buffer = data.mPreLFO.mForcedBufferSeconds;
+  const CCoord minibuffer = data.mPreLFO.mMinibufferSeconds;
+
+  // zoom out when a single forced buffer exceeds the default window
+  while (forced_buffer > window_sec) {
```

Comment:
I see that immediately above you seem to be using the pre-LFO data, but I nonetheless see in the plugin that the scale will change while the buffer size changes via LFO. I didn't try to debug... any ideas? IMO it's better if the scale stays the same so that the "animation" feels monotonic, but it's not that bad (and of course there are always going to be settings that push the limits of this simple visualizer).

Try for example:
9.5 ms/buffer
6.30 divisor
buffer size LFO 0.28 Hz, 100% depth, sine

Reply:
Oh shoot, it is the data source that is wrong! The "pre-LFO" mini-buffer size was pre- its own LFO, but post- the forced buffer size's LFO. I have changed it to now be pre-both. Good catch thanks!